src/hash/rescue/rpx/tests.rs

```diff
@@ -0,0 +1,188 @@
+use alloc::{collections::BTreeSet, vec::Vec};
+
+use proptest::prelude::*;
+use rand_utils::rand_value;
+
+use super::{Felt, Hasher, Rpx256, StarkField, ZERO};
+use crate::{hash::rescue::RpxDigest, ONE};
+
+#[test]
+fn hash_elements_vs_merge() {
+    let elements = [Felt::new(rand_value()); 8];
+
+    let digests: [RpxDigest; 2] = [
+        RpxDigest::new(elements[..4].try_into().unwrap()),
+        RpxDigest::new(elements[4..].try_into().unwrap()),
+    ];
+
+    let m_result = Rpx256::merge(&digests);
+    let h_result = Rpx256::hash_elements(&elements);
+    assert_eq!(m_result, h_result);
+}
+
+#[test]
+fn merge_vs_merge_in_domain() {
+    let elements = [Felt::new(rand_value()); 8];
+
+    let digests: [RpxDigest; 2] = [
+        RpxDigest::new(elements[..4].try_into().unwrap()),
+        RpxDigest::new(elements[4..].try_into().unwrap()),
+    ];
+    let merge_result = Rpx256::merge(&digests);
+
+    // ------------- merge with domain = 0
+    // ----------------------------------------------------------
+
+    // set domain to ZERO. This should not change the result.
+    let domain = ZERO;
+
+    let merge_in_domain_result = Rpx256::merge_in_domain(&digests, domain);
+    assert_eq!(merge_result, merge_in_domain_result);
+
+    // ------------- merge with domain = 1
+    // ----------------------------------------------------------
```

Comment:
Let's shorten this separator to avoid wrapping.

Reply:
Fixed